src/finalizer/finalizer.cpp

```diff
@@ -492,12 +492,64 @@ extern "C" HRESULT DetectSdk(LPWSTR sczSdkFeatureBandVersion, LPWSTR sczArchitec
     return hr;
 }
 
+void RemoveInstallStateFile(LPWSTR sczSdkFeatureBandVersion, LPWSTR sczPlatform)
+{
+    HRESULT hr = S_OK;
+    LPWSTR sczProgramData = NULL;
+    LPWSTR sczInstallStatePath = NULL;
+    LPWSTR sczPath = NULL;
+
+    hr = ShelGetFolder(&sczProgramData, CSIDL_COMMON_APPDATA);
+    ExitOnFailure(hr, "Failed to get shell folder.");
+
+    hr = PathConcat(sczProgramData, L"dotnet", &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat dotnet to install state path.");
+
+    hr = PathConcat(sczInstallStatePath, L"workloads", &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat workloads to install state path.");
+
+    hr = PathConcat(sczInstallStatePath, sczPlatform, &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat platform (%ls) to install state path.", sczPlatform);
+
+    hr = PathConcat(sczInstallStatePath, sczSdkFeatureBandVersion, &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat feature band (%ls) to install state path.", sczSdkFeatureBandVersion);
+
+    hr = PathConcat(sczInstallStatePath, L"installstate", &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat installstate to install state path.");
+
+    hr = PathConcat(sczInstallStatePath, L"default.json", &sczInstallStatePath);
+    ExitOnFailure(hr, "Failed to concat default.json to install state path.");
```

Comment:
Eventually we plan to also have files in this folder that keep track of global.json files that pin to a specific workload set.  We might want to clean those up too in the future (though it's not as critical, as they shouldn't cause issues if they are left there and you reinstall an SDK from the feature band).

Reply:
We can always add more files. It will be a lot simpler once the repo merge is done to keep things in sync.